nbtlib/path.py

```diff
@@ -0,0 +1,265 @@
+        for accessor in new_accessors:
+            accessors = extend_accessors(accessors, accessor)
+
+        return self.from_accessors(accessors)
+
+    def __add__(self, other):
+        if isinstance(other, Path):
+            return self[other]
+        elif isinstance(other, str):
+            return self[Path(other)]
+        else:
+            return NotImplemented
+
+    def __radd__(self, other):
+        if isinstance(other, Path):
+            return other[self]
+        elif isinstance(other, str):
+            return Path(other)[self]
+        else:
+            return NotImplemented
+
+    def __eq__(self, other):
+        if isinstance(other, str):
+            other = Path(other)
+        return super().__eq__(other)
+
+    def __ne__(self, other):
+        if isinstance(other, str):
+            other = Path(other)
+        return super().__ne__(other)
+
+    def __hash__(self):
+        return super().__hash__()
+
+    @classmethod
+    def from_accessors(cls, accessors=()):
+        return super().__new__(cls, accessors)
+
+    def traverse(self, tag):
+        tags = [(None, tag)]
+
+        setter = None
+        deleter = None
+
+        for accessor in self:
+            setter = getattr(accessor, 'set', setter)
+            deleter = getattr(accessor, 'delete', deleter)
+
+            tags = accessor.get(tags)
+
+        return tags, setter, deleter
+
+    def get(self, tag):
+        return [tag for _, tag in self.traverse(tag)[0]]
+
+    def set(self, tag, value):
+        tags, setter, _ = self.traverse(tag)
+
+        if setter:
+            setter(tags, value)
+
+    def delete(self, tag):
+        tags, _, deleter = self.traverse(tag)
+
+        if deleter:
+            deleter(tags)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({str(self)!r})'
+
+    def __str__(self):
+        segments = ['']
+
+        for accessor in self:
+            segment = str(accessor)
+
+            if not segment or segment.startswith('['):
+                segments[-1] += segment
+
+            elif segment.startswith('{'):
+                if segments[-1].endswith('[]'):
+                    segments[-1] = segments[-1][:-2] + f'[{segment}]'
+                else:
+                    segments[-1] += segment
+
+            else:
+                segments.append(segment)
+
+        return '.'.join(filter(None, segments))
+
+
+def parse_accessors(path):
+    try:
+        parser = Parser(tokenize(path))
+    except InvalidLiteral:
+        return ()
+
+    while True:
+        try:
+            tag = parser.parse()
+        except InvalidLiteral as exc:
+            raise InvalidPath(f'Invalid path at position {exc.args[0][0]}') from exc
+
+        if isinstance(tag, String):
+            if parser.current_token.type == 'QUOTED_STRING':
+                yield NamedKey(tag[:])
+            else:
+                yield from (NamedKey(key) for key in tag.split('.') if key)
+
+        elif isinstance(tag, List):
```

Comment:
`List` token is a bit different from List in NBT.
`[    ]` , `[  5  ]` ,`[   {}   ]`  are not allowed.
but
`[00005]`,`[{         }]`are allowed.

Reply:
This is kind of the same as above, I think it's useful to allow `[ ]` , `[ 5 ]`, `[ {} ]` and `[{ }]` as long as they turn into something minecraft understands when you call `str()` on the path.

And thanks for noticing `[00005]` I'm going to fix that :+1:

Reply:
Alright things like `[00005]` should now work properly 4efdcc1